lib/codecs/src/encoding/format/csv.rs

```diff
@@ -223,7 +225,7 @@ impl Encoder<Event> for CsvSerializer {
             if fields_written > 0 {
                 loop {
```

Comment:
(optional) 💬 

The code looks fine to me. I was thinking for a way to extract the three `loop {}` into a common method but it is not straightforward. Perhaps we can extract each `loop {}` into a separate method.

Reply:
Agreed. Previously I (before using `enumerate()` on the fields) I failed to convince the compiler to agree with my function design. But I am learning quite a bit about rusts borrowing at the moment so I will give it another try.

Reply:
[c730d58](https://github.com/vectordotdev/vector/pull/18320/commits/c730d58d6c7d69aa897278766accd513bd023e01) is my shot on the mentioned refactor. Personally I feel it actually adds complexity and I would like to revert it... (or improve the implementation) your opinion @pront ?

Reply:
I agree with you, please revert. It was worth a try, thanks.